lib/matplotlib/animation.py

```diff
@@ -874,6 +874,12 @@ def __init__(self, fig, event_source=None, blit=False):
         if self._blit:
             self._setup_blit()
 
+        # Saves the animation is the figure class to prevent Python's garbage
+        # collection from deleteing the animation.
+        if (not hasattr(fig, 'animations')):
```

Comment:
I have added the animations attribute in figure __init__ instead of the hasattr.